crates/sui-indexer/src/models_v2/objects.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use diesel::prelude::*;
+use sui_types::digests::ObjectDigest;
+
+use move_bytecode_utils::module_cache::GetModule;
+use sui_types::base_types::{ObjectID, ObjectRef, SequenceNumber};
+use sui_types::dynamic_field::{DynamicFieldInfo, DynamicFieldType};
+use sui_types::object::Object;
+use sui_types::object::{ObjectFormatOptions, ObjectRead};
+
+use crate::errors::IndexerError;
+use crate::schema_v2::objects;
+use crate::types_v2::IndexedObject;
+
+// NOTE: please add updating statement like below in pg_indexer_store_v2.rs,
+// if new columns are added here:
+// objects::epoch.eq(excluded(objects::epoch))
+#[derive(Queryable, Insertable, Debug, Identifiable, Clone, QueryableByName)]
+#[diesel(table_name = objects, primary_key(object_id))]
+pub struct StoredObject {
+    pub object_id: Vec<u8>,
+    pub object_version: i64,
+    pub object_digest: Vec<u8>,
+    pub checkpoint_sequence_number: i64,
+    pub owner_type: i16,
+    pub owner_id: Option<Vec<u8>>,
+    pub serialized_object: Vec<u8>,
+    pub coin_type: Option<String>,
+    // TODO deal with overflow
+    pub coin_balance: Option<i64>,
```

Comment:
even if the balance is stored as i64, we could just treat it as a bit pattern and straight cast to u64

Reply:
i64 is enough to handle max SUI coin given 10B total amount, indeed an issue for other coins.